### DISCUSSION TITLE
Single App vs Multiple Apps

### DISCUSSION THREAD

**Discussion by tcf909** [uNetworking/uWebSockets.js#489] [Q&A] [answered]
Hello,

I can't say enough great things about your project. Definitely a breath of fresh air. 

I wanted to get your thoughts on running a single App with many websocket connections, vs splitting the websocket connections across multiple App instances (in the same thread) -- this with regard to performance, events, queueing, latency, etc.

Is there a great benefit in splitting up websocket connections across multiple App instances?

Thanks

**Comment by ghost** [comment, marked as answer]
Hello, you typically have 1 app per thread. There is no benefit to having many apps in the same thread.

**Reply by tcf909** [reply]
just confirming: even when it comes to things like tcp buffers, congestion windows, etc?

**Reply by ghost** [reply]
You can read the user manual
